### ISSUE TITLE
Offsets for a specific array

### ISSUE THREAD

**Issue by kelloggm** [kelloggm/checker-framework#76]
Consider the following code:

```java
void test(@IndexFor("a") int start, @IndexOrHigh("a") int end, int[] a) {
      a[end - start];
}
```

If we know that end > start, always, then this code is correct. But we don't have a way to express that.

**Comment by kelloggm** [comment]
Likely to be closed by implementing the proposal in #75

**Comment by smillst** [comment]
Fixed via https://github.com/typetools/checker-framework/commit/4c3163a48582f8837fdbfe1176043a273e378ddf
